str/reviewer_runner.py

```diff
@@ -0,0 +1,106 @@
+
+SAMTOOLS_IMAGE = config['images']['samtools']
+REVIEWER_IMAGE = config['images']['reviewer']
+
+
+@click.option('--locus', help='Locus identifier as per catalog')
+@click.option('--catalog', help='GCP path to catalog')
+@click.option('--input-dir', help='GCP path to input-dir, includes gs://')
+@click.argument('cpg-sample-ids', nargs=-1)
+@click.command()
+def main(locus, catalog, input_dir, cpg_sample_ids: list[str]):
+
+    # Initializing Batch
+    b = get_batch()
+    ref = b.read_input_group(
+        base=REF_FASTA,
+        catalog=catalog,
+        fai=REF_FASTA + '.fai',
+        dict=REF_FASTA.replace('.fasta', '').replace('.fna', '').replace('.fa', '')
+        + '.dict',
+    )
+    for cpg_id in cpg_sample_ids:
+        # read in bam file from EH output
+        bam_input = b.read_input(os.path.join(input_dir, f'{cpg_id}_eh.realigned.bam'))
+
+        # read in VCF from EH output
+        vcf_input = b.read_input(os.path.join(input_dir, f'{cpg_id}_eh.vcf'))
+
+        # BAM file must be sorted and indexed prior to inputting into REViewer
+        samtools_job = b.new_job(name=f'Sorting and indexing {cpg_id}')
+        samtools_job.image(SAMTOOLS_IMAGE)
+        file_size_bytes = (
+            AnyPath(os.path.join(input_dir, f'{cpg_id}_eh.realigned.bam'))
+            .stat()
+            .st_size
+        )
+        padding = 5
+        file_size_gib = str(math.ceil(file_size_bytes / (1024**3)) + padding) + 'G'
+        samtools_job.storage(file_size_gib)
+        samtools_job.declare_resource_group(
+            bam={
+                'sorted.bam': '{root}.sorted.bam',
+                'sorted.bam.bai': '{root}.sorted.bam.bai',
+            }
+        )
+
+        samtools_job.command(
+            f"""
+        echo "sorting {bam_input}";
+        samtools sort -@ 4 {bam_input} -o {samtools_job.bam['sorted.bam']};
+
+        echo "indexing {samtools_job.bam['sorted.bam']}";
+        samtools index -@ 4 {samtools_job.bam['sorted.bam']} -b -o {samtools_job.bam['sorted.bam.bai']};
+
+        """
+        )
+
+        # REViewer job
+        reviewer_job = b.new_job(name=f'Visualising {locus} for {cpg_id}')
+        reviewer_job.image(REVIEWER_IMAGE)
+        reviewer_job.depends_on(samtools_job)
+
+        reviewer_job.declare_resource_group(
+            ofile={
+                'svg': '{root}.' + f'{locus}.svg',
+                'metrics.tsv': '{root}.metrics.tsv',
+                'phasing.tsv': '{root}.phasing.tsv',
+            }
+        )
+
+        reviewer_job.command(
+            f"""
+            ./REViewer-v0.2.7-linux_x86_64 --reads {samtools_job.bam['sorted.bam']} \\
+            --vcf {vcf_input} --reference {ref.base} --catalog {ref.catalog} \\
+            --out {reviewer_job.ofile} --locus {locus}
+            """
+        )
+    # output writing
+    reviewer_output_path = output_path(f'{locus}/{cpg_id}_{locus}_reviewer', 'analysis')
+    b.write_output(reviewer_job.ofile, reviewer_output_path)
```

Comment:
These should be inside the loop, otherwise you're only writing one output from all jobs. This is what the `undefined-loop-variable` was warning you against.